### DISCUSSION TITLE
Deploy RustDesk client with complete configuration via GPO in AD environment

### DISCUSSION THREAD

**Discussion by schoolandme** [rustdesk/rustdesk#5781] [Q&A]
I setup my own server in windows. Now I want to push rustdesk client to all PC, i tried few ways still unable to push with complete configuration. My config is just host and key only, below is the way i tried: 

1) Due to GPO Software deployment only support MSI installer, I use MSI Wrapper to generate a MSI installer then I rename it like host=x.x.x.x, key=xxxxxxxx.msi, it will install with blank config, is not the config I want. I tried different argument or cmd like --config or --import-config also didn't work.

2) I tried use the powershell script to deploy but will have UAC issue since domain user do not have the right to install and obviously I can't make everyone to have admin right to install.

If anybody have experienced this issue before please advices me because have tested many weeks. Thanks very much all.

**Comment by ukdt7** [comment]
Maybe you should try a deployment solution with a scripthost. We use opsi for example. it needs a bit of learning but once you have it, it is a pretty good way to deploy software. we run a powershellscript before the logon as a system user with opsi.

**Reply by schoolandme** [reply]
is Opsi free or license software?

**Reply by My1** [reply]
we are trying with opsi but there are some problems with making a script, like the setup basically giving control back despite not being finished, likely it forks into another process and exits itself, similar for the uninstaller

**Reply by kproske** [reply]
Hello My1,
You have to set a Timeout in your OPSI-Script. Then you can rollout Rustdesk  via opsi

**Comment by rustdesk** [comment]
> 1. Due to GPO Software deployment only support MSI installer, I use MSI Wrapper to generate a MSI installer then I rename it like host=x.x.x.x, key=xxxxxxxx.msi, it will install with blank config, is not the config I want. I tried different argument or cmd like --config or --import-config also didn't work.

can you share how do you wrap? i may give some suggestion.

**Comment by rustdesk** [comment]
MSI custom client is ready now, https://twitter.com/rustdesk/status/1782250852574769663

**Reply by RafaelSoaresP** [reply]
Can't access X on my end. Is there any complimentary documentation to this?
